### PR TITLE
Unbreak building C++ with -DLMMS_DEBUG

### PR DESCRIPTION
**Pull request by softrabbit**
Small fixes for building with the -DLMMS_DEBUG flag defined for C++

### CODE REVIEW

**Comment by Lukas-W** [comment]
Ah yeah, `-DLMMS_DEBUG` is missing in `CMAKE_CXX_FLAGS_DEBUG`. Could you add it after [`CMakeLists.txt:373`](https://github.com/softrabbit/lmms/blob/81c04db5af32e1aa8093bfd4bdc4cd9b0c2f547f/CMakeLists.txt#L373)?

File: src/core/SampleRecordHandle.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@
 #include "Pattern.h"
 #include "SampleBuffer.h"
 #include "SampleTrack.h"
-
+#include "debug.h"
```
Comment (Lukas-W) [inline src/core/SampleRecordHandle.cpp]: As the assert macro in `debug.h` doesn't do anything if `LMMS_DEBUG` is not defined, the '#ifdef' check in line 119 can be removed.

**Comment by softrabbit** [comment]
Looks like windows builds are broken? I'll have to dig into details later.

**Comment by Lukas-W** [comment]
Yes but that's not your fault, see #1565

**Comment by softrabbit** [comment]
Oops. Looks like cherry-picking the CMakelist.txt changes into my branch complicated this a bit. Should I redo this in a cleaner way?

**Comment by Lukas-W** [comment]
Don't worry, I'll merge it manually.

**Pull request closed without merging**